### PR TITLE
action.yml: support OAuth clients in addition to authkeys

### PR DESCRIPTION
**Pull request by DentonGentry**
Fixes https://github.com/tailscale/github-action/issues/59

### CODE REVIEW

**Comment by DentonGentry** [comment]
I'd especially like suggestions on the names of arguments, as we can basically never change them once merged.

```yaml
  - name: Tailscale
    uses: tailscale/github-action@v2
    with:
      oauth_client_id: ${{ secrets.TS_OAUTH_CLIENT_ID }}
      oauth_secret: ${{ secrets.TS_OAUTH_SECRET }}
      oauth_tags: tag:ci
```

The action doesn't really need the client-id, but if we don't provide a way to pass it I think we'll get a never-ending series of issues filed asking what people are supposed to do with the client-id. The argument is marked as optional and not checked for. If omitted, the action will still run.

The oauth_tags argument could conceivably be named just `tags`, but it does have to match the Tags configured in the OAuth client.

File: action.yml [hunk 1]
```diff
@@ -9,11 +9,21 @@ branding:
 inputs:
   authkey:
     description: 'Your Tailscale authentication key, from the admin panel.'
-    required: true
+    required: false
+    deprecationMessage: 'An OAuth API endpoint https://tailscale.com/s/oauth-clients is recommended instead of an authkey'
```
Comment (willnorris) [inline action.yml:13]: s/endpoint/client/ ?

**Review by willnorris — APPROVED** [review]
suggestion on "tags" name, otherwise LGTM

File: action.yml [hunk 2]
```diff
@@ -9,11 +9,21 @@ branding:
...
+  oauth_client_id:
+    description: 'Your Tailscale OAuth Client ID.'
+    required: false
+  oauth_secret:
+    description: 'Your Tailscale OAuth Client Secret.'
+    required: false
+  oauth_tags:
```
[the remark this answers is not in the document]
Comment (DentonGentry) [reply action.yml]: Changed name of the input to just `tags` and added a note that the OAuth client must have permission to apply that Tag.

File: action.yml (hunk 1 above)
Comment (DentonGentry) [reply action.yml]: Done.

**Comment by DentonGentry** [comment]
Good observation about underscores vs dashes, updated the inputs to use dashes.

**Pull request merged**